lib/RequestRouter.js

```diff
@@ -514,8 +520,8 @@ RequestRouter.prototype.addPropertyError = function(property, errorCode) {
         });
     }
 
-    if (this._request.getMethod() == Request.METHOD_POST && this.isCollection()) {
-        //remove the appended items@ if posting to a collection
+    if (!!~[Request.METHOD_PATCH, Request.METHOD_POST, Request.METHOD_PUT, Request.METHOD_DELETE].indexOf(this.getMethod()) && this.isCollection()) {
```

Comment:
We also want to patch and put stuff for collections as well... we actually have a PATCH use case alreayd in our code that uses that